### PR TITLE
SOC-1445 Don't allow token-less GETs to add comments

### PR DESCRIPTION
**Pull request by garthwebb**
Note this is based on the SOC-1444 branch.

https://wikia-inc.atlassian.net/browse/SOC-1445

### CODE REVIEW

**Comment by armonr** [comment]
LGTM

File: extensions/wikia/ArticleComments/classes/ArticleCommentsAjax.class.php [hunk 1]
```diff
@@ -169,9 +169,16 @@ static public function axReply() {
 	static public function axPost() {
 		global $wgRequest, $wgUser, $wgLang;
 
+		$result = [ 'error' => 1 ];
+
+		try {
+			$wgRequest->isValidWriteRequest( $wgUser );
+		} catch ( \BadRequestException $bre ) {
+			return $result;
```
Comment (kvas-damian) [inline extensions/wikia/ArticleComments/classes/ArticleCommentsAjax.class.php:178]: For better UI there should be an error message displayed for user with info what happened and what to do now. There is already translation key that can be reused: `sessionfailure`

File: extensions/wikia/ArticleComments/js/ArticleComments.js [hunk 2]
```diff
@@ -446,13 +447,11 @@
 						}
 					}
 
-					//readd events
+					//re-add events
 					ArticleComments.addHover();
-					//clear error box
-					$('#article-comm-info').html('');
```
Comment (garthwebb) [inline extensions/wikia/ArticleComments/js/ArticleComments.js:452]: This article-comm-info element has `display: none` by default, and shows up, unstyled, at the top of the page.  It doesn't seem like it was ever finished.  Using BannerNotifications instead.

**Comment by kvas-damian** [comment]
LGTM

**Pull request merged**